### PR TITLE
Adding rpc support for lists. Fix #1144

### LINKED ISSUE #1144: rpc blocks can't accept list inputs

### PR DESCRIPTION
**Pull request by ellienguyen**

### CODE REVIEW

File: src/client/threads.js [hunk 1]
```diff
@@ -318,13 +319,47 @@ NetsProcess.prototype.parseRPCResult = function (result) {
     return result;
 };
 
+function listToJSON(list) {
```
Comment (brollb) [inline src/client/threads.js]: I don't really like these as global functions - we should find somewhere better to declare them... It's not immediately clear where a better place would be though

File: src/client/threads.js [hunk 2]
```diff
@@ -318,13 +319,47 @@ NetsProcess.prototype.parseRPCResult = function (result) {
...
+    if (input.charAt(0) !== '[') {
+        return input;
+    }
+    return JSON.parse(input).map(parseList);
+}
+
+function toQueryString(obj, prefix) {
+    var str = [], k, v;
+    for(var p in obj) {
+        if (!obj.hasOwnProperty(p)) {continue;} // skip things from the prototype
```
Comment (brollb) [inline src/client/threads.js]: using `Object.keys(obj)` then using a `for` loop should be faster and should be a little more legible

File: src/client/threads.js [hunk 3]
```diff
@@ -318,13 +319,47 @@ NetsProcess.prototype.parseRPCResult = function (result) {
...
+        if (~p.indexOf('[')) {
```
Comment (brollb) [inline src/client/threads.js]: It might be better to use `> -1` for legibility (rather than using the "+1 and change sign" operator).

Comment (brollb) [inline src/client/threads.js]: Also when do you run into this case?

File: src/client/threads.js (hunk 1 above)
Comment (ellienguyen) [inline src/client/threads.js]: Maybe we can move it into a `utils.js` so that it's not global anymore and easy to reuse

**Comment by brollb** [comment]
Thanks, @ellienguyen!

**Pull request merged**